metagraph/src/annotation/taxonomy/label_to_taxid.cpp

```diff
@@ -0,0 +1,95 @@
+        logger->error("Can't determine the type of the given label {}. Please make sure that the labels are in a recognized format.", label);
+        std::exit(1);
+    }
+}
+
+bool TaxonomyBase::get_taxid_from_label(const std::string &label, TaxId *taxid) const {
+    if (label_type == TAXID) {
+        *taxid = static_cast<uint32_t>(std::stoull(utils::split_string(label, "|")[1]));
+        return true;
+    } else if (TaxonomyBase::label_type == GEN_BANK) {
+        std::string acc_version = get_accession_version_from_label(label);
+        if (not accversion_to_taxid_map.count(acc_version)) {
+            return false;
+        }
+        *taxid = accversion_to_taxid_map.at(acc_version);
+        return true;
+    }
+
+    logger->error("Run get_taxid_from_label() for unknown label {}.", label);
+    std::exit(1);
+}
+
+std::string TaxonomyBase::get_accession_version_from_label(const std::string &label) const {
+    if (label_type == TAXID) {
+        return utils::split_string(utils::split_string(label, "|")[2], " ")[0];
+    } else if (label_type == GEN_BANK) {
+        return utils::split_string(label, "|")[3];;
+    }
+
+    logger->error("Run get_accession_version_from_label() for unknown label {}.", label);
+    std::exit(1);
+}
+
+// TODO improve this by parsing the compressed ".gz" version (or use https://github.com/pmenzel/taxonomy-tools)
+void TaxonomyBase::read_accversion_to_taxid_map(const std::string &filepath,
+                                                const graph::AnnotatedDBG *anno_matrix = NULL) {
+    std::ifstream f(filepath);
+    if (!f.good()) {
+        logger->error("Failed to open accession to taxid map table {}", filepath);
+        exit(1);
+    }
+
+    std::string line;
+    getline(f, line);
+    if (!utils::starts_with(line, "accession\taccession.version\ttaxid\t")) {
+        logger->error("The accession to taxid map table is not in the standard (*.accession2taxid) format {}.", filepath);
+        exit(1);
+    }
+
+    tsl::hopscotch_set<std::string> input_accessions;
+    if (anno_matrix != NULL) {
+        for (const std::string &label : anno_matrix->get_annotation().get_all_labels()) {
+            input_accessions.insert(get_accession_version_from_label(label));
+        }
+    }
+
+    while (getline(f, line)) {
+        if (line == "") {
+            logger->error("The accession to taxid map table contains empty lines. Please make sure that this file was not manually modified {}.", filepath);
+            exit(1);
+        }
+        std::vector<std::string> parts = utils::split_string(line, "\t");
+        if (parts.size() <= 2) {
+            logger->error("The accession to taxid map table contains incomplete lines. Please make sure that this file was not manually modified {}.", filepath);
+            exit(1);
+        }
+        if (input_accessions.size() == 0 || input_accessions.count(parts[1])) {
+            accversion_to_taxid_map[parts[1]] = static_cast<TaxId>(std::stoull(parts[2]));
```

Comment:
this will be casted to TaxId automatically
```suggestion
            accversion_to_taxid_map[parts[1]] = std::stoull(parts[2]);
```